src/main/java/org/opensearch/sdk/handlers/EnvironmentSettingsResponseHandler.java

```diff
@@ -11,27 +11,43 @@
 import org.apache.logging.log4j.Logger;
 import org.opensearch.env.EnvironmentSettingsResponse;
 import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.settings.Settings;
 import org.opensearch.sdk.ExtensionsRunner;
 import org.opensearch.threadpool.ThreadPool;
 import org.opensearch.transport.TransportException;
 import org.opensearch.transport.TransportResponseHandler;
 
 import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
 
 /**
  * This class handles the response from OpenSearch to a {@link ExtensionsRunner#sendEnvironmentSettingsRequest} call.
  */
 public class EnvironmentSettingsResponseHandler implements TransportResponseHandler<EnvironmentSettingsResponse> {
     private static final Logger logger = LogManager.getLogger(EnvironmentSettingsResponseHandler.class);
 
+    private CountDownLatch inProgressLatch;
+    private Settings environmentSettings;
```

Comment:
This should be initialized to something (probably `Settings.EMPTY`) to return in case you get an exception or we are interrupted.